### DISCUSSION TITLE
how to use Riverpod with searchDelegate ?

### DISCUSSION THREAD

**Discussion by Aboidrees** [rrousselGit/riverpod#2236] [Q&A] [answered]
Context:
I am using search delegate `showSearch(context: context, delegate: CustomeSearchDelegate);` to search Geo locations.
I want to save former searches, so the user can research some locations. And this will be in the top of the suggestions (same as YouTube search). 
`CustomeSearchDelegate()` is a configuration object passed to a stateful widget (`_SearchPage(delegate:delegate)`) that handles the search UI, which belongs to Flutter SDK.

My Problem:
  
How to use Riverpod providers to show cached (local) or saved (remote server) suggestion?

**Comment by praisegeek** [comment, marked as answer]
```dart
 // inside SearchDelegate
 @override
  Widget buildResults(BuildContext context) {
         // Create a RiverPod ConsumerWidget 
         return SearchResultView(query: query)
  }
  
  // Access state, api services etc.. e.g ref.read(appProvider.notifier).getLocations(query) 
  class SearchResultView extends ConsumerWidget {
     ...
  }
  ```
